R/modelling.R

```diff
@@ -172,24 +173,12 @@ fit_seromodel <- function(serodata,
   )
 
   if (class(seromodel_fit@sim$samples) != "NULL") {
-    seromodel_object <- list(
-      seromodel_fit = seromodel_fit,
-      serodata = serodata,
-      stan_data = stan_data,
-      exposure_years = exposure_years,
-      exposure_ages = exposure_ages
-    )
+    seromodel_object <- seromodel_fit
+    return(seromodel_object)
   } else {
-    seromodel_object <- list(
-      seromodel_fit = "no model",
-      serodata = serodata,
-      stan_data = stan_data,
-      exposure_years = exposure_years,
-      exposure_ages = exposure_ages
-    )
+    seromodel_object <- "no model"
+    return(seromodel_object)
```

Comment:
Could you add a comment explaining how this situation can arise please?

Reply:
A trivial example for this would be to have `n_iters = 1`. In general, any set of inputs yielding to an error in config_argss when running `rstan::sampling` above, like for instance choosing `n_thin > n_iters`.

Reply:
According to the docs, shouldn't this be `seromodel_fit@mode != 0`?

https://mc-stan.org/rstan/reference/stanfit-class.html

> An integer indicating the mode of the fitted model. 0 indicates sampling mode, 1 indicates test gradient mode (no sampling is done), and 2 indicates error mode (an error occurred before sampling). Most methods for stanfit objects are useful only if mode=0.

https://mc-stan.org/rstan/reference/stanmodel-method-sampling.html

> [Returns] an object of S4 class stanfit representing the fitted results. Slot mode for this object indicates if the sampling is done or not.

Reply:
Yes! Thanks for pointing this out. I think this is way cleaner.